### PR TITLE
cli/client: add getnamestate and getnameresource

### PR DESCRIPTION
**Pull request by tynes**
This PR adds two new commands to the `hsd-cli` tool.

- `$ hsd-cli namestate [name]`
- `$ hsd-cli resource [name]`

It also adds the correct methods to the `NodeClient` so that the cli can have this functionality. I find these methods to be useful when exploring the chainstate from the cli. The methods currently wrap the RPC endpoints so it is possible to get the same functionality with `$ hsd-cli rpc ...`, but making the commands first class citizens will help to make it easier for people using the cli tool, since namestate and resources are important in Handshake. Potentially consider switching to using the HTTP endpoints once they are merged in.

### CODE REVIEW

File: bin/hsd-cli [hunk 1]
```diff
@@ -64,6 +64,33 @@ class CLI {
...
+    const name = this.config.str(0, '');
+
+    const ns = await this.client.getNameState(name);
+
+    if (!ns){
+      this.log('NameState not found.');
+      return;
+    }
+
+
```
Comment (pinheadmz) [inline bin/hsd-cli]: nit: extra line

File: bin/hsd-cli [hunk 2]
```diff
@@ -64,6 +64,32 @@ class CLI {
...
+    this.log(ns);
+  }
+
+  async getNameResource() {
+    const name = this.config.str(0, '');
+
+    const resource = await this.client.getNameResource(name);
+
+    if (!resource){
+      this.log('Resouce not found.');
```
Comment (pinheadmz) [inline bin/hsd-cli]: nit / sp `Resource`

File: bin/hsd-cli [hunk 3]
```diff
@@ -64,6 +64,32 @@ class CLI {
     this.log(info);
   }
 
+  async getNameState() {
+    const name = this.config.str(0, '');
+
+    const ns = await this.client.getNameState(name);
+
+    if (!ns){
+      this.log('NameState not found.');
```
Comment (pinheadmz) [inline bin/hsd-cli]: Hm, camel case in error messages for the CLI user...? I dunno, the command is all lowercase `namestate`.
Not sure how I feel about it. Although also worth noting, that even a totally random name will return _something_:

```
{
  "start": {
    "reserved": false,
    "week": 31,
    "start": 1152
  },
  "info": null
}
```

So might be moot anyway.

File: bin/hsd-cli (hunk 2 above)
Comment (tynes) [reply bin/hsd-cli]: Good catch

**Comment by tynes** [comment]
Looking back at this, I think we should consider if the RPC endpoints `getnameinfo` and `getnameresource` return all of the information that a consumer would want because the other CLI methods use the REST interface and there isn't a REST interface for these 2 commands yet. If anything is missing from the RPC interface, we can get away with implementing the REST interface in a different way and not have any breaking changes. Then this PR could add those methods and have them call the REST interface.

**Comment by pinheadmz** [comment]
I don't think I realized back in October this was wrapping existing rpc commands. What's the use case? A cli command already exists: `hsd-rpc getnameresource`

**Comment by tynes** [comment]
I'm closing this because I want to get feedback from users on what is missing from the RPC methods so that we can add useful HTTP endpoints.

**Pull request closed without merging**